src/libtls/tls_server.c

```diff
@@ -261,6 +262,35 @@ static bool peer_supports_curve(private_tls_server_t *this,
 	return FALSE;
 }
 
+/**
+ * TLS 1.3 key exchange key share
+ */
+typedef struct {
+	uint16_t curve;
+	chunk_t key_share;
+} key_share_t;
+
+/**
+ * Check if peer sent a key share of a given TLS named DH group
+ */
+static bool peer_offered_curve(array_t *key_shares, tls_named_group_t curve,
+							   key_share_t *out)
+{
+	key_share_t peer;
+	int i;
+
+	for (i = 0; i < array_count(key_shares); i++)
+	{
+		array_get(key_shares, i, &peer);
+		if (curve == peer.curve)
+		{
+			*out = peer;
```

Comment:
You could make this assignment optional, so you can pass NULL to check if the peer offered a particular curve.